packages/ai-bot/lib/responder.ts

```diff
@@ -222,7 +228,14 @@ export class Responder {
       ).cancel();
       this.sendMessageEvent();
     }
-    await Promise.all(this.messagePromises);
+
+    let results = await Promise.all(this.messagePromises);
+
+    for (let result of results) {
+      if (result && 'errorMessage' in result) {
+        await this.onError(result.errorMessage);
+      }
+    }
   }
```

Comment:
I think this is fine. It could also be modeled with `Promise.allSettled` which means that a message promise could reject if it fails, which may be a little more expressive.

Reply:
I think even with using `allSettled`, we'd still need individual `.catch` handler on each promise, otherwise the unhandled rejection will still crash the process (I tested this). So even with `allSettled` it would look very similar. I think this would be a clearer option if we were interested in all results values but here we're just checking if there was an error.